cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -347,6 +355,25 @@ struct LockedGroup {
     page: CatalogPage,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Compose {
+    /// The composing environment's manifest that was on disk at lock-time.
+    pub composer: Manifest,
+    /// Metadata and manifests for the included environments in the order
+    /// that they were specified in the composing environment's manifest.
+    pub include: Vec<LockedInclude>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct LockedInclude {
+    pub manifest: Manifest,
+    pub descriptor: IncludeDescriptor,
+    // TODO: once we consider remote environments, add this field
+    // pub remote: Option<RemoteSource>
```

Comment:
> Imagine I need to decide where to look up an environment. I have to match on descriptor in order to get the local path if there is one. If there isn't, I find out that it's remote and need to now match on source. I think it's more natural to model that as an Option instead, unless we foresee composing environments that exist somewhere other than on disk and on FloxHub.

note nonblocking: so instead of having a type tag we'll just use the existence of `Option` 🤔 I think that's fine just double checking you don't think we should ever have a tag for anything

Reply:
I'm not exactly sure what you mean by "we should never have a tag for anything". There are two places where you would identify an include as "remote": once in the descriptor and once when this `source` field is `Some(_)`. Aside: it's not optimal that it's technically possible to have `IncludeDescriptor::Local` and `source = Some(_)`, but all of the proposed options have that problem.

The options I proposed are:
- Model it as an option because we don't think we'll have include types other than "remote on Floxhub specifically" and "anything local".
- Model it in a less natural way but allow for more flexibility in the future

I struggle to think of what we would need that flexibility for, but going from `Option` -> `enum` would be a breaking change.

Reply:
I was just asking because the ticket has `#[serde(tag = "type")]` and `"type": "remote"`

> I struggle to think of what we would need that flexibility for, but going from `Option` -> `enum` would be a breaking change.

YAGNI, I think an `Option` is good for now 👍